cartographer/pose_graph/optimizer/optimizer.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_POSE_GRAPH_OPTIMIZER_OPTIMIZER_H_
+#define CARTOGRAPHER_POSE_GRAPH_OPTIMIZER_OPTIMIZER_H_
+
+#include "cartographer/pose_graph/pose_graph_data.h"
+
+namespace cartographer {
+namespace pose_graph {
+
+class Optimizer {
+ public:
+  Optimizer() = default;
+  virtual ~Optimizer() = default;
+
+  Optimizer(const Optimizer&) = delete;
+  Optimizer& operator=(const Optimizer&) = delete;
+
+  virtual ceres::Solver::Summary Solve(PoseGraphData* data) const = 0;
```

Comment:
I think we might have to abstract here from ceres since this base class is intended for using it with other optimizers than ceres, if I'm not mistaken. 
How about an enum class returning whether it converged or not?